### DISCUSSION TITLE
How can I add an exclusion filter in the suborgs?

### DISCUSSION THREAD

**Discussion by justinmchase** [github/safe-settings#466] [Q&A]
For example suppose I have a suborg file such as;

```yml
suborgrepos:
  - example-*
```

But I want it to match all repos except for ones such as `example-*-test`? 

I tried this but it doesn't seem to work:

```yml
suborgrepos:
  - example-*
  - !example-*-test
```

I can't seem to find any documentation about support for exclusion globs. Is this supported? Am I doing it wrong?

**Comment by martinm82** [comment]
This would be really a great feature next to the exclude for teams and collaborators where include/exclude are supported.

**Reply by martinm82** [reply]
@decyjphr is this something we could easily add into the `Glob` class? There is some code already that checks the suborgrepo item whether it matches a regex: https://github.com/github/safe-settings/blob/main-enterprise/lib/settings.js#L267.

**Reply by justinmchase** [reply]
Uff, you're implementing glob yourself:
https://github.com/github/safe-settings/blob/main-enterprise/lib/glob.js

Consider using the glob package instead:
https://www.npmjs.com/package/glob

**Reply by decyjphr** [reply]
@martinm82 Yes, that could be done. Probably a trivial change.

@justinmchase I remember looking at that. I think that npm module is used to get a list of files using a Glob pattern. Our class is just a wrapper to do searches in a string for glob pattern-like matches. Probably a different name would have been better.

**Reply by justinmchase** [reply]
Yeah I suppose you're right, I haven't looked this up in a while but I guess glob uses minimatch internally for the glob filtering of files:

https://github.com/isaacs/node-glob/issues/530#issuecomment-1552085158
